### PR TITLE
Disassembler script to help diff assembly between versions

### PR DESCRIPTION
**Pull request by cadmic**
Summary: the goal is get a human-readable diff just by running the `diff` command. This requires spimdisasm flags to avoid writing anything that depends on global addresses or specific versions. Furthermore `--name-vars-by-file` requires disassembling everything at once instead of file-by-file.

Things that may or may not need to be fixed before merge:
* Output is written to e.g. `oot/cpmd/asm` instead of separate `oot/cpmd/asm/text` and `/oot/cpmd/asm/data` directories. This is because I often ran e.g. `diff oot/{cpm,cpmd}/asm/text/code/z_play.text.s` in my shell, and if I then wanted to inspect the data diff I would have to edit the middle of the path instead of just the extension.
* No support for `--split-functions`, `--nuke-pointers`,  `--function-info`, or `--add-filename`
* spimdisasm flags `--save-context`, `--variables`, and `--constants` are not hooked up in the Makefile. I didn't use these but maybe `--variables` is at least good to add (though I'd like a way to turn it off)
* Animal forest overlays are probably broken. The kind of work, but e.g. `dnm/jp/asm/ovl_submenu/ovl_submenu.text.s` starts with
```
dlabel D_file_007778B0.data_000000
  addiu       $sp, $sp, -0x60
  sw          $ra, 0x24($sp)
  sw          $a0, 0x60($sp)
  sw          $a1, 0x64($sp)
glabel D_file_007778B0.data_000010
  sw          $a2, 0x68($sp)
glabel D_file_007778B0.data_000014
  lw          $t6, 0x68($sp)
  lui         $at, (0x3F800000 >> 16)
  lw          $t7, 0x64($sp)
  beqz        $t6, D_file_007778B0.data_000090
   lw         $t1, 0x60($sp)
```
and I'm not sure why. It seems like the .text segment gets processed as a .data segment for some reason.
* No support for faster disassembly via `disasm_list`
* `makerom` is not disassembled
* Probably needs spimdisasm release?
* Need to update README (though after we decide what should be changed)

### CODE REVIEW

File: mips/ZeldaTables.py [hunk 1]
```diff
@@ -34,8 +34,10 @@ def __init__(self, filename, vromStart, vromEnd, romStart, romEnd, sizeVrom, com
         self.number = number
 
 
-def getFileAddresses(filePath: Path | None) -> dict[str, FileAddressesEntry]:
-    table: dict[str, FileAddressesEntry] = dict()
+
+
+def getFileAddressList(filePath: Path | None) -> list[FileAddressesEntry]:
+    result: list[FileAddressesEntry] = []
```
Comment (AngheloAlf) [inline mips/ZeldaTables.py:40]: I'm curious on why you changed this instead of using `.values()` on the dict

**Review by AngheloAlf — COMMENTED** [review]
The main issue I can see with this PR is it isn't a drop-in replacement for the already existing stuff.
Maybe wired everything up so you can choose between the current behavior or the new behavior when disassembling?
Maybe a make target like `make simple-diff` (or a better name lol)

File: mips/ZeldaTables.py (hunk 1 above)
Comment (cadmic) [reply mips/ZeldaTables.py:40]: I needed to get the next entry for animal forest overlays, and while technically Python dictionaries are now always sorted by insertion order it still feels weird to depend on that, idk

**Comment by cadmic** [comment]
Abandoning for now

**Pull request closed without merging**